### PR TITLE
Update 'julia-config.jl' for static AOT Julia compilation

### PR DESCRIPTION
**Pull request by lucatrv**
Add flags required for static AOT Julia compilation, and '--all' option to output all flags.
See [static-julia](https://github.com/JuliaComputing/static-julia) project.

### CODE REVIEW

**Comment by ViralBShah** [comment]
The `--all` flag is a good idea.

File: contrib/julia-config.jl [hunk 1]
```diff
@@ -36,7 +37,7 @@ end
 function ldflags()
     fl = "-L$(shell_escape(libDir()))"
     if Sys.iswindows()
-        fl = fl * " -Wl,--stack,8388608"
+        fl = fl * " -Wl,--export-all-symbols -Wl,--stack,8388608"
```
Comment (vtjnash) [inline contrib/julia-config.jl]: Can you open an issue for this and assign me? I know we used to emit a correct file, so it seems like that has regressed.

**Comment by lucatrv** [comment]
I closed this PR and opened a new one #24140, sorry for the inconvenience

**Comment by yuyichao** [comment]
>
> I closed this PR and opened a new one #24140
> <https://github.com/JuliaLang/julia/pull/24140>, sorry for the
> inconvenience
>

Please just update the existing PR whenever possible. It's much easier to
track discussions that way.

**Comment by lucatrv** [comment]
Yes, sorry, my fault

**Pull request closed without merging**